src/main/java/edu/cornell/kfs/pdp/service/impl/CuPaymentFileServiceImpl.java

```diff
@@ -159,6 +164,25 @@ private void assignDisbursementTypeCode(PaymentGroup pg) {
         }
     }
     
+    /**
+     * This method sends error emails in a manner similar to that from PaymentInputFileType.parse(),
+     * except that our code will also attempt to send an email to the customer if possible.
+     */
+    private void sendErrorEmailForGenericException(
+            Exception genericException, String incomingFileName, PaymentFileLoad paymentFile) {
+        PaymentFileLoad paymentFileForEmail;
+        try {
+            paymentFileForEmail = getCustomerProfileFromUnparsableFile(incomingFileName, paymentFile);
+        } catch (RuntimeException e) {
+            paymentFileForEmail = paymentFile;
```

Comment:
should we log this exception?

Reply:
should paymentFileForEmail be set to something if the parse fails?

Reply:
Ok, I'll add some logging here just in case.

However, I don't understand what you mean about "paymentFileForEmail be set to something if the parse fails". If the plain-text parse fails, it'll get set to the "paymentFile" value that was passed into the method, though "paymentFile" could be null in such a case. When the email-sending code gets called, it will automatically ignore the passed-in "paymentFileForEmail" value if it's null, so I don't think we need to worry about a NullPointerException here.

If there's some other potential problem here that you meant to point out, please let me know.

Reply:
Thanks for the further explanation, I think that answers the concern I was trying to mention.